### PR TITLE
Add gsoci.azurecr.io as push target

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/roadmap/issues/2895

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -33,9 +33,11 @@ const (
 	customizedImagesFile = "images/customized-images.yaml"
 	defaultPlatform      = "linux/amd64"
 	dockerTransport      = "docker://"
-	quayURL              = "quay.io/giantswarm"
-	aliyunURL            = "giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm"
 	filteredFileSuffix   = ".filtered"
+
+	aliyunURL = "giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm"
+	azureURL  = "gsoci.azurecr.io"
+	quayURL   = "quay.io/giantswarm"
```
Comment (piontec) [inline main.go:40]: for the time being, we also need to push to the old ACR, `giantswarm.azurecr.io`

Comment (piontec) [reply main.go:40]: Can we also make this configurable? We will have to change it a few times in the coming months, recompiling each time doesn't seem very appealing...

File: main.go [hunk 2]
```diff
@@ -119,10 +122,13 @@ func (img *CustomImage) RetagUsingSHA() error {
 	if len(img.DockerfileExtras) == 0 {
 		source := fmt.Sprintf("%s%s@sha256:%s", dockerTransport, img.Image, img.SHA)
 		wg := sync.WaitGroup{}
-		wg.Add(2)
+		wg.Add(3)
 		// Quay
 		destination := fmt.Sprintf("%s%s/%s:%s", dockerTransport, quayURL, destinationName, destinationTag)
 		go copyImage(&wg, errorCounter, source, destination)
+		// AzureCR
```
Comment (piontec) [inline main.go:129]: Shouldn't we use `oci://` transport now?

Comment (kubasobon) [reply main.go:129]: I don't think so, no.

File: main.go (hunk 1 above)
Comment (kubasobon) [reply main.go:40]: Sure, I can make it more configurable.

Comment (kubasobon) [reply main.go:40]: Actually, it would require similar effort. Let's talk offline

File: main.go [hunk 3]
```diff
@@ -33,9 +33,11 @@ const (
 	customizedImagesFile = "images/customized-images.yaml"
 	defaultPlatform      = "linux/amd64"
 	dockerTransport      = "docker://"
-	quayURL              = "quay.io/giantswarm"
-	aliyunURL            = "giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm"
 	filteredFileSuffix   = ".filtered"
+
+	aliyunURL = "giantswarm-registry.cn-shanghai.cr.aliyuncs.com/giantswarm"
+	azureURL  = "gsoci.azurecr.io"
```
Comment (marians) [inline main.go:39]: Should there be the `/giantswarm` suffix after `gsoci.azurecr.io`?

Comment (kubasobon) [reply main.go:39]: It should! Thank you!

**Pull request merged**